File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.cpp
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(int balls, int share) {
    share = min(share, balls - share);
    if (share == 0) return 1;
    long long res = solution(balls - 1, share - 1);
    return res * balls / share;
}

File: 프로그래머스/0/120840. 구슬을 나누는 경우의 수/구슬을 나누는 경우의 수.java
class Solution {
    public int solution(int balls, int share) {
        int k = Math.min(share, balls - share);
        long answer = 1;
        for (int i = 1; i <= k; ++i) {
            answer = answer * (balls - k + i) / i;
        }
        return (int) answer;
    }
}